src/carts/src/main/java/mushop/carts/Cart.java

```diff
@@ -49,10 +50,24 @@ public boolean removeItem(String itemId) {
     }
 
     public void merge(Cart cart) {
-        this.customerId = cart.getCustomerId();
```

Comment:
Why remove this.customerId = cart.getCustomerId()?
I thought this handles case where user starts a cart before logging in and then logs in and starts adding more items.